anaDACScan.py

```diff
@@ -0,0 +1,318 @@
+    
+    if len(args.infilename.split('/')) > 1 and len(args.infilename.split('/')[len(args.infilename.split('/')) - 2].split('.')) == 5:
+        scandate = args.infilename.split('/')[len(args.infilename.split('/')) - 2]
+    else:    
+        scandate = 'noscandate'
+        
+    for oh in ohArray:
+        if scandate == 'noscandate':
+            os.system("mkdir -p "+ elogPath+"/"+chamber_config[oh])
+        else:
+            os.system("mkdir -p "+ dataPath+"/"+chamber_config[oh]+"/dacScans/"+scandate)
+            
+    # Determine which DAC was scanned and against which ADC
+    nameX = ""
+    nameY = ""
+    for event in dacScanFile.dacScanTree:
+        nameX = str(event.nameX.data())
+        nameY = str(event.nameY.data())
+        break # all entries will be the same
+
+    if nameY not in ['ADC0', 'ADC1']:
+        print("Error: unexpected value of nameY: '%s'"%nameY)
+        exit(1)
+
+    from utils.anaInfo import nominalDacValues    
+        
+    if nameX not in nominalDacValues.keys():
+        print("Error: unexpected value of nameX: '%s'"%nameX)
+        exit(1)
+
+    nominal_iref = nominalDacValues['CFG_IREF'][0]
+
+    #convert all currents to uA 
+    if nominalDacValues['CFG_IREF'][1] == 'A':
+        nominal_iref *= pow(10.0,6)        
+    elif nominalDacValues['CFG_IREF'][1] == 'mA':
+        nominal_iref *= pow(10.0,3)
+    elif nominalDacValues['CFG_IREF'][1] == 'uA':
+        pass
+    elif nominalDacValues['CFG_IREF'][1] == 'nA':
+        nominal_iref *= pow(10.0,-3)
+    else:
+        print("Error: unexpected units for nominal reference current: '%s'"%nominalDacValues['CFG_IREF'][1])
+        exit(1)
+     
+    nominal = nominalDacValues[nameX][0]
+
+    #convert all voltages to mV
+    if nominalDacValues[nameX][1] == "V" or nominalDacValues[nameX][1] == "A":
+        nominal *= pow(10.0,3)
+    elif nominalDacValues[nameX][1][0] == 'm':
+        pass
+    elif nominalDacValues[nameX][1][0] == 'u':
+        nominal *= pow(10.0,-3)
+    elif nominalDacValues[nameX][1][0] == 'n':
+        nominal *= pow(10.0,-6)
+    else:
+        print("Error: unexpected units: '%s'"%nominalDacValues[nameX][1])
+        exit(1)
+
+    calInfo = {}
+
+    if args.calFileList != None:
+        for line in open(args.calFileList):
+            if line[0] == "#":
+                continue
+            line = line.strip(' ').strip('\n')
+            first = line.split(' ')[0].strip(' ')
+            second = line.split(' ')[1].strip(' ')
+            tuple_calInfo = parseCalFile(second)
+            calInfo[int(first)] = {'slope' : tuple_calInfo[0], 'intercept' : tuple_calInfo[1]}
+
+    #for each OH, check if calibration files were provided, if not search for the calFile in the $DATA_PATH, if it is not there, then skip that OH for the rest of the script
+    for oh in ohArray:
+        if oh not in calInfo.keys():
+            calDacCalFile = "{0}/{1}/calFile_calDac_{1}.txt".format(dataPath,chamber_config[oh])
```

Comment:
why `calDAC`?  The cal file is not for the `CFG_CAL_DAC` register.  We are calibrating the ADC, e.g.:

```python
`{0}/{1}/calFile_{2}_{1}.txt".format(dataPath,chamber_config[ohN],nameY)
```